### ISSUE TITLE
2018.3.0 changed Windows minion names on upgrade

### ISSUE THREAD

**Issue by morganwillcock** [saltstack/salt#47174]
### Description of Issue/Question

It seems that my Windows minions are being renamed, and force me to re-accept the key on the master.

### Steps to Reproduce Issue
I've had a couple of quick goes at trying to reproduce it, but haven't managed it for the moment. It did change of the name of every Windows minion I had though (and Linux minions were fine). The arguments for the installer (from the Task Scheduler) were:
<pre>/s /c ""c:\salt\var\cache\salt\minion\extrn_files\base\repo.saltstack.com\windows\Salt-Minion-2018.3.0-Py2-AMD64-Setup.exe" /S /master=salt01 /minion-id=%COMPUTERNAME%"</pre>

Normally I just use the hostname with no domain name, but after upgrade everything comes back as hostname.fdqn and I have to re-accept all the keys and fix all the logic in the name matching.

### Versions Report
<pre># salt --versions-report
Salt Version:
           Salt: 2018.3.0
 
Dependency Versions:
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.2
      docker-py: Not Installed
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.9 (default, Jun 29 2016, 13:08:31)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: debian 8.10 
         locale: UTF-8
        machine: x86_64
        release: 3.16.0-5-amd64
         system: Linux
        version: debian 8.10</pre>

**Comment by gtmanfred** [comment]
I am unable to replicate this.

@dwoz @twangboy Have yall heard anything about this?

Thanks,
Daniel

**Comment by twangboy** [comment]
The installation has changed a bit in 2018.3. See here for details:
https://docs.saltstack.com/en/latest/topics/installation/windows.html#silent-installer-options

Basically, the installer will no longer modify an existing installation. It will only modify a custom config or the default config contained in the installer. That's for GUI operation.

The command line has two new options: `/default-config` and `/custom-config` which behave as they do in the gui. However, if you pass the `/master=` or the `/minion-name=` options, it will use the default config. If you passed the `/custom-config` option then the custom config will be used.

So, basically, to perform an upgrade without messing with the existing config all you need to pass is the `/S`. I would guess that the `%COMPUTERNAME%` environment variable contains a FQDN and that's what's being written to the default config.

Another option would be to specify the `/master=` but not specify a minion name. By default salt will used the name of the machine if not defined in the minion config.

**Comment by morganwillcock** [comment]
> I would guess that the %COMPUTERNAME% environment variable contains a FQDN and that's what's being written to the default config.

@twangboy %COMPUTERNAME% is the short version without the domain, but are you sure the problem isn't that the switch name has been changed? It used to be `/minion-id` (not sure why I'd be using it otherwise) and is now `/minion-name`. Was that intentional?

**Comment by twangboy** [comment]
@morganwillcock Oh, so it was probably the reverse. If you were passing `/minion-id` then that was being ignored and it was just using whatever was in the existing config. If it was a new install, then it was just using the hostname.

In 2018.3, when you define `/master=` or `/minion-name=`, the config is being overwritten... either by the default or a custom. So, maybe that's what happened.

Anyway, for upgrades it is recommended that you just run the installer with `/S` to keep the current config untouched.

**Comment by morganwillcock** [comment]
@twangboy thanks, I think that explains it, as I was passing the master name and then (mistakenly) nothing else.
